server/conn.go

```diff
@@ -190,3 +191,7 @@ func (c *Conn) UnsetStatus(status uint16) {
 func (c *Conn) HasStatus(status uint16) bool {
 	return c.status&status > 0
 }
+
+func (c *Conn) SetWarnings(warnings uint16) {
```

Comment:
Do you really need this method?

Reply:
Yes, that is the only way to set the unexported warnings parameter on the Conn struct. If I want to tell clients connecting to my server there were warnings executing their queries, I have to be able to manipulate that parameter. Alternatively, the warnings field could be exported, but that is little consistent with for instance `status`, which also has a somewhat similar setter function `SetStatus`.

Reply:
snippet from my HandleQuery code:
```golang
func (q MyHandler) HandleQuery(query string) (*mysql.Result, error) {
    // ...

    result, err := backendConn.Execute(query)
    clientConn.SetWarnings(result.Warnings)

    // ...
    return result, err
}
```